ddtrace/tracer.py

```diff
@@ -1045,18 +1052,18 @@ def shutdown(self, timeout=None):
         """
         with self._shutdown_lock:
             # Thread safety: Ensures tracer is shutdown synchronously
+            try:
```

Comment:
Why do we need to move this up? Can we leave comment here

Reply:
some other parts of the code crash during shutdown. I could fix it, but I don't think it's in the scope of this PR